### PR TITLE
chore: Adds github CI workflow

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

This lib is actively used by the SMF and it currently did not contain any github CI workflow validation. This PR adds minimal validations:
- Builds the project
- Runs unit tests

This PR also adds a dependabot file for automated dependency updates.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Canonical Ltd.
```
Comment (gab-arrobo) [inline .github/dependabot.yml:2]: 2023 or 2024?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        run: go build
+
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.mod"
```
Comment (gab-arrobo) [inline .github/workflows/master.yml:34]:
```suggestion
          go-version-file: 'go.mod'
```

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Canonical Ltd.
```
Comment (gab-arrobo) [inline .github/workflows/master.yml:2]: 2023 or 2024?

**Review by gab-arrobo — COMMENTED** [review]
Just in case, are you planning to open a PR to bump up GO version?

File: .github/dependabot.yml (hunk 1 above)
Comment (gruyaume) [reply .github/dependabot.yml]: Fixed

File: .github/workflows/master.yml (hunk 3 above)
Comment (gruyaume) [reply .github/workflows/master.yml]: Fixed

**Review by gab-arrobo — APPROVED** [review]
+1

**Comment by gab-arrobo** [comment]
> Yes this should probably be done but in a different PR. And adding Dependabot will come with a bunch of other updates as well.

It would be good to open a PR bumping up GO version before dependabot starts creating PRs to update the "old" dependencies

**Comment by gab-arrobo** [comment]
ok to test

**Pull request merged**